### PR TITLE
Add Kubernetes overview to docs

### PR DESCRIPTION
**Pull request by kerin**
Intended to give a very brief but useful overview of what Kubernetes is, how it works, what the major resource types are, what extra resource types and services we've added, and some links to resources elsewhere - any additions to the _Further Kubernetes Information_ section gratefully received.

### CODE REVIEW

File: source/documentation/index.md [hunk 1]
```diff
@@ -2,10 +2,71 @@
...
+
+### Further Kubernetes Information
+
+<!--
+Content based on https://github.com/ministryofjustice/cloud-platform-user-guide/blob/main/source/documentation/concepts/kubernetes.html.md.erb
+-->
+
+Here are some links to introductory Kubernetes resources:
+
+ * For a brief and light-hearted introduction to what Kubernetes is all about, try this [comic][k8s-comic] from Google.
```
Comment (jonathanglassman) [inline source/documentation/index.md:57]: Recommend removing "light-hearted" as best practice is to not have subjective terms like this, as different users may have different definitions of this

File: source/documentation/index.md [hunk 2]
```diff
@@ -2,10 +2,71 @@
...
+* [Pods](https://kubernetes.io/docs/concepts/workloads/pods/) - a group of one ore more containers
+* [Deployments](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/) - one or more replicas of a pod, and an associated [deployment strategy](https://www.weave.works/blog/kubernetes-deployment-strategies)
+* [Services](https://kubernetes.io/docs/concepts/services-networking/service/) - a means of exposing pods or deployments as a network service via a DNS name
+* [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) - a means of exposing Services externally via a load balancer, with associated SSL/TLS certificates and DNS names
+* [ConfigMap](https://kubernetes.io/docs/concepts/configuration/configmap/) - a collection of configuration key/value pairs, commonly used to provide environment variables to containers
+* [Secrets](https://kubernetes.io/docs/concepts/configuration/secret/) - much like ConfigMap, but for smaller, sensitive key/value pairs
+
+### Custom kubernetes objects
+
+The GOV.UK Kubernetes Platform includes some third party API objects and controllers to allow for easy integration with AWS services:
```
Comment (jonathanglassman) [inline source/documentation/index.md:42]: Recommend removing "easy" as best practice is to not have subjective terms like this, as different users may have different definitions of this

**Review by jonathanglassman — CHANGES_REQUESTED** [review]
Looks good overall, made some comments, happy to discuss

File: source/documentation/index.md [hunk 3]
```diff
@@ -2,10 +2,71 @@
...
+- Self-healing deployments
+- Secrets and config management
+
+See [What is Kubernetes?](https://kubernetes.io/docs/concepts/overview/what-is-kubernetes/) for more details.
+
+### Kubernetes declarative state model
+
+Declarative configuration is a fundamental part of the design of Kubernetes, and it is important to understand the distinction between this declarative model and more traditional imperative models.
+
+Briefly, the desired state of an application is submitted to the Kubernetes API (e.g. this container at a specific tag should be running, with these environment variables, with this port exposed to the internet), and Kubernetes internal components will then work to bring the current state of the application in line with the declared desired state; this is known as **state reconciliation**, and is handled via [Kubernetes controllers](https://kubernetes.io/docs/concepts/architecture/controller/).
```
[the remark this answers is not in the document]
Comment (kerin) [reply source/documentation/index.md]: This section has been rewritten for clarity, now with examples of declarative vs. imperative with a link to a more detailed blog post on the subject.

File: source/documentation/index.md [hunk 4]
```diff
@@ -2,10 +2,71 @@
...
+
+### Kubernetes objects
+
+The Kubernetes API supports many [object types](https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/) representing the state of component parts of an application. The major resource types that will be used in most (if not all) applications are:
+
+* [Pods](https://kubernetes.io/docs/concepts/workloads/pods/) - a group of one ore more containers
+* [Deployments](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/) - one or more replicas of a pod, and an associated [deployment strategy](https://www.weave.works/blog/kubernetes-deployment-strategies)
+* [Services](https://kubernetes.io/docs/concepts/services-networking/service/) - a means of exposing pods or deployments as a network service via a DNS name
+* [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) - a means of exposing Services externally via a load balancer, with associated SSL/TLS certificates and DNS names
```
[the remark this answers is not in the document]
Comment (kerin) [reply source/documentation/index.md]: These terms are all proper nouns and are always written in title case, and are also direct references to specific Kubernetes objects, so all k8s objects have been titled cased and formatted as code for consistency.

[the remark this answers is not in the document]
Comment (kerin) [reply source/documentation/index.md]: All users should know what these terms mean, but I've linked the terms to the relevant wikipedia pages, and removed the reference to the now-deprecated SSL.

**Comment by jonathanglassman** [comment]
@kerin looks good to me, thank you for making the changes so quickly

**Pull request merged**